include/aspect/material_model/depth_dependent.h

```diff
@@ -0,0 +1,129 @@
+/*
+  Copyright (C) 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#ifndef __aspect__model_depth_dependent_h
+#define __aspect__model_depth_dependent_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <deal.II/base/parsed_function.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    namespace internal
+    {
+      class RadialViscosityLookup;
+    }
+
+    template <int dim>
+    class DepthDependent : public MaterialModel::Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Initialization function. Loads the material data and sets up
+         * pointers.
+         */
+        virtual
+        void
+        initialize ();
+
+        /**
+         * Function to compute the material properties in @p out given the
+         * inputs in @p in.
+         */
+
+        virtual
+        void
+        evaluate (const typename Interface<dim>::MaterialModelInputs &in,
+                  typename Interface<dim>::MaterialModelOutputs &out) const;
+
+        static void
+        declare_parameters (ParameterHandler &prm);
+
+        virtual void
+        parse_parameters (ParameterHandler &prm);
+
+        virtual bool
+        viscosity_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        virtual bool
+        density_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        virtual bool
+        compressibility_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        virtual bool
+        specific_heat_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        virtual bool
+        thermal_conductivity_depends_on (const NonlinearDependence::Dependence dependence) const;
+
+        virtual bool is_compressible () const;
+
+        virtual double reference_viscosity () const;
+
+        virtual double reference_density () const;
+
+      private:
+        enum
+        {
```

Comment:
here and for the member functions below, please add small piece of documentation explaining what they do, what their arguments are, and what they return. same for the member variables.